### ISSUE TITLE
[v2] Defining inbound-cidrs on multiple Ingresses do not merge correctly

### ISSUE THREAD

**Issue by matt-demers** [kubernetes-sigs/aws-alb-ingress-controller#1486]
Trying to create 2 Ingress objects with the same group name. Both ingresses define the same annotation for `alb.ingress.kubernetes.io/inbound-cidrs: "10.0.0.0/8, 100.64.0.0/10"`.

This outputs the following error in the controller logs
```
{"level":"error","ts":1602349807.374569,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"ingress","name":"group1","namespace":"","error":"failed to merge listPort config for port: 443: conflicting sslPolicy, /: [10.0.0.0/8 100.64.0.0/10], [] | test2/test: [10.0.0.0/8 100.64.0.0/10], []","errorVerbose":"conflicting sslPolicy, /: [10.0.0.0/8 100.64.0.0/10], [] | test2/test: [10.0.0.0/8 100.64.0.0/10], []\nsigs.k8s.io/aws-load-balancer-controller/pkg/ingress.(*defaultModelBuildTask).mergeListenPortConfigs\n\t/workspace/pkg/ingress/model_builder.go:221\nsigs.k8s.io/aws-load-balancer-controller/pkg/ingress.(*defaultModelBuildTask).run\n\t/workspace/pkg/ingress/model_builder.go:166\nsigs.k8s.io/aws-load-balancer-controller/pkg/ingress.(*defaultModelBuilder).Build\n\t/workspace/pkg/ingress/model_builder.go:104\nsigs.k8s.io/aws-load-balancer-controller/controllers/ingress.(*GroupReconciler).Reconcile\n\t/workspace/controllers/ingress/group_controller.go:85\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:233\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:209\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:188\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:90\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1374\nfailed to merge listPort config for port: 443\nsigs.k8s.io/aws-load-balancer-controller/pkg/ingress.(*defaultModelBuildTask).run\n\t/workspace/pkg/ingress/model_builder.go:168\nsigs.k8s.io/aws-load-balancer-controller/pkg/ingress.(*defaultModelBuilder).Build\n\t/workspace/pkg/ingress/model_builder.go:104\nsigs.k8s.io/aws-load-balancer-controller/controllers/ingress.(*GroupReconciler).Reconcile\n\t/workspace/controllers/ingress/group_controller.go:85\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:233\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:209\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:188\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:90\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1374","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:235\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:209\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:188\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:90"}
```

Using image `amazon/aws-alb-ingress-controller:v2.0.0-rc2`

Applying the first ingress works correctly, and the security group contains the correct inbound rules. Applying the second ingress causes the error. Removing the annotation `alb.ingress.kubernetes.io/inbound-cidrs` from the second ingress fixes the error. Other properties like the `ssl-policy` do seem to merge correctly.

The ingress definitions:
```yaml
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: test
  namespace: test
  annotations:
    kubernetes.io/ingress.class: alb
    alb.ingress.kubernetes.io/group.name: group1
    alb.ingress.kubernetes.io/scheme: internal
    alb.ingress.kubernetes.io/ssl-policy: ELBSecurityPolicy-TLS-1-2-2017-01
    alb.ingress.kubernetes.io/healthcheck-protocol: "HTTP"
    alb.ingress.kubernetes.io/backend-protocol: "HTTP"
    alb.ingress.kubernetes.io/listen-ports: "[{\"HTTPS\":443}]"
    alb.ingress.kubernetes.io/inbound-cidrs: "10.0.0.0/8, 100.64.0.0/10"
    alb.ingress.kubernetes.io/healthcheck-path: /
spec:
  tls:
    - hosts:
      - test1.example.com
  rules:
    - host: test1.example.com
      http:
        paths:
          - path: /
            backend:
              serviceName: nginx-service
              servicePort: 80
```

```yaml
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: test
  namespace: test2
  annotations:
    kubernetes.io/ingress.class: alb
    alb.ingress.kubernetes.io/group.name: group1
    alb.ingress.kubernetes.io/scheme: internal
    alb.ingress.kubernetes.io/ssl-policy: ELBSecurityPolicy-TLS-1-2-2017-01
    alb.ingress.kubernetes.io/healthcheck-protocol: "HTTP"
    alb.ingress.kubernetes.io/backend-protocol: "HTTP"
    alb.ingress.kubernetes.io/listen-ports: "[{\"HTTPS\":443}]"
    alb.ingress.kubernetes.io/inbound-cidrs: "10.0.0.0/8, 100.64.0.0/10"
    alb.ingress.kubernetes.io/healthcheck-path: /
spec:
  tls:
    - hosts:
      - test2.example.com
  rules:
    - host: test2.example.com
      http:
        paths:
          - path: /
            backend:
              serviceName: nginx-service
              servicePort: 80
```

**Comment by M00nF1sh** [comment]
@matt-demers 
Hi Matt,
Thanks for reporting this, will be fixed with https://github.com/kubernetes-sigs/aws-alb-ingress-controller/pull/1484

**Comment by M00nF1sh** [comment]
closing this since it's addressed by v2.0.0-rc3 release [https://github.com/kubernetes-sigs/aws-alb-ingress-controller/releases/tag/v2.0.0-rc3] :D.
feel free to reopen if further issue exists.
